instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractor.java

```diff
@@ -54,7 +54,15 @@ public static <REQUEST, RESPONSE> HttpClientAttributesExtractorBuilder<REQUEST,
   @Override
   public void onStart(AttributesBuilder attributes, Context parentContext, REQUEST request) {
     super.onStart(attributes, parentContext, request);
-    internalSet(attributes, SemanticAttributes.HTTP_URL, getter.url(request));
+    internalSet(attributes, SemanticAttributes.HTTP_URL, stripSensitiveData(getter.url(request)));
+  }
+
+  private static String stripSensitiveData(String url) {
+    if (url.isEmpty() || url == null) {
+      return url;
+    }
+    // replace username & password
+    return url.replaceFirst("(?<=\\/\\/)(.+)(?=@)", "username:password");
```

Comment:
Regex pattern matching is rather expensive; even more so when you're not using a precompiled `Pattern`.

How about using `indexOf()` to find out the positions of `//` and `@` instead?
Also, you could just simply remove the credentials (together with the `@` character); this would be more in line with the spec:
> `http.url MUST NOT`contain credentials passed via URL in form of `https://username:password@www.example.com/`. In such case the attribute's value should be `https://www.example.com/`